src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -28711,6 +28740,76 @@ public static class S
                 Diagnostic(ErrorCode.ERR_BadArgType, "A _").WithArguments("2", "out A", "out B").WithLocation(8, 18)
                 );
         }
+
+        [Fact]
+        public void DeclarationInLocalFunctionParameterDefault()
+        {
+            var text = @"
+class C
+{
+    public static void Main(int arg)
+    {
+        void Local2(bool b = M(M(out int z1), z1), int s2 = z1) {}
```

Comment:
>void Local2(bool b = M(M(out int z1), z1), int s2 = z1) {} [](start = 8, length = 58)

Consider adding a test where a variable declared within a default parameter value is referenced within the local function that owns the parameter.